tests/TestRun.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Tests;
+
+use Tests\Support\AssertionException;
+use Tests\Support\AssertionExceptionExpectException;
+use Exception;
+
+class BaseTest
+{
+    public function __construct($repositoryName, $fileName)
+    {
+        $this->dirName = getenv('JSON_STORAGE_PATH');
+        $this->fileName = $fileName;
+        $this->repositoryName = $repositoryName;
+    }
+
+    public function run()
+    {
+        $testsCount = 0;
+        $completedCount = 0;
+        $keyword = 'test';
+        $length = strlen($keyword);
+
+        foreach (get_class_methods($this) as $method) {
+            if (str_starts_with($method, $keyword)) {
+                $this->beforeTestsProcessing();
+                $testsCount++;
+
+
+                echo $this->fileName . substr($method, $length) . ': ' . PHP_EOL;
+
+                try {
+                    $this->{$method}();
+                } catch (AssertionException $error) {
+                    echo $error->getMessage();
+
+                    continue;
+                }
+
+                $completedCount++;
+
+                echo 'Success.' . PHP_EOL . PHP_EOL;
+            }
+        }
+
+        $methodsFail = $testsCount - $completedCount;
+
+        echo 'Total tests run: ' . $testsCount . PHP_EOL . 'Completed: ' . $completedCount . PHP_EOL . 'Failed: ' . $methodsFail . PHP_EOL . PHP_EOL;
+
+        if ($methodsFail > 0 && getenv('APP_ENV') === 'tests_runner') {
+            exit(1);
+        }
+    }
+
+    protected function beforeTestsProcessing()
+    {
+        if (file_exists("tests/Dumb/{$this->fileName}_perfect_value.json")) {
+
+            $data = $this->getDataSet("tests/Dumb/{$this->fileName}_perfect_value.json");
```

Comment:
as we discussed please store non one perfect value, but different files with data